dev/archery/archery/cli.py

```diff
@@ -900,22 +900,16 @@ def release_cherry_pick(obj, version, dry_run, recreate):
     """
     Cherry pick commits.
     """
-    from .release import Release, MinorRelease, PatchRelease
+    from .release import Release
 
     release = Release.from_jira(version, jira=obj['jira'], repo=obj['repo'])
-    if not isinstance(release, (MinorRelease, PatchRelease)):
-        raise click.UsageError('Cherry-pick command only supported for minor '
-                               'and patch releases')
 
     if not dry_run:
         release.cherry_pick_commits(recreate_branch=recreate)
```

Comment:
How about moving `dry_run` mode implementation to `Release`?

```suggestion
    release.cherry_pick_commits(recreate_branch=recreate, dry_run=dry_run)
```

`Release` knows what `git` command lines should be ran. If we move `dry_run` mode implementation to `Release`, we can implement `dry_run` mode and non `dry_run` mode nearby.

Reply:
Good idea, but may handle it in follow-up PR.